src/util/http/HttpClient.cpp

```diff
@@ -211,3 +227,36 @@ HttpOrHttpsResponse sendHttpOrHttpsRequest(
     return sendRequest.operator()<HttpsClient>();
   }
 }
+
+// ____________________________________________________________________________
+template <typename StreamType>
+cppcoro::generator<std::string> HttpClientImpl<StreamType>::readWebSocketStream(
+    std::unique_ptr<HttpClientImpl<StreamType>> client, std::string_view host,
+    std::string_view target) {
+  AD_CORRECTNESS_CHECK(client->stream_);
+  client->ws_ = std::make_unique<boost::beast::websocket::stream<StreamType>>(
+      std::move(*(client->stream_)));
+  client->ws_->handshake(host, target);
+
+  beast::flat_buffer buffer;
+  for (;;) {
+    client->ws_->read(buffer);
+    co_yield beast::buffers_to_string(buffer.data());
+    buffer.consume(buffer.size());
+  }
+}
+
+// ____________________________________________________________________________
+cppcoro::generator<std::string> readHttpOrHttpsWebsocketStream(
+    const ad_utility::httpUtils::Url& url, std::string_view target) {
+  auto listen = [&]<typename Client>() -> cppcoro::generator<std::string> {
```

Comment:
Have a look at the  util/typeIdentity.h header, there i have a trick that makes the templated lambdas easier to call.